mars/services/task/supervisor/processor.py

```diff
@@ -588,6 +601,83 @@ def get_tileable_details(self):
             }
         return tileable_infos
 
+    def get_tileable_subtasks(self, tileable_id: str):
+        tileable_id_to_tileable = dict()
+        tileable_to_subtasks = dict()
+        subtask_results = dict()
+        returned_subtasks = set()
+        default_result = SubtaskResult(progress=0.0, status=SubtaskStatus.pending)
+
+        subtask_list = []
+        dependency_list = []
+
+        for processor in self._task_id_to_processor.values():
+            for stage in processor.stage_processors:
+                tileable_id_to_tileable.update(stage.tileable_id_to_tileable)
+                tileable_to_subtasks.update(stage.tileable_to_subtasks)
```

Comment:
Just lookup inside these dicts instead of merging them. The latter will be expensive when graphs are very large.